### PR TITLE
Include support for periodic data acquisition mode.

### PR DESCRIPTION
**Pull request by WoefulDerelict**
This expands the driver with support for periodic data acquisition mode and control of the sensor's measurement parameters. Support for periodic data acquisition mode is necessary to enable and explore the device's alert feature.

### CODE REVIEW

**Comment by WoefulDerelict** [comment]
In a recent response to a request for information Sensirion's field applications engineer has stated that the SHT3X-D data stream is a FIFO stack. I will attempt to verify this shortly and update if necessary.

**Comment by WoefulDerelict** [comment]
The field applications engineer I have been in contact with appears to be mistaken. Utilising a simple test programme that monitored the data output in periodic data acquisition mode while the device's internal heater switched on I was able to identify a clear FILO trend in the returned results and contacted the engineer with my findings.

**Review by tannewt — COMMENTED** [review]
Hi! Thanks for adding this. Would you mind adding an example that does periodic sampling to better show the API? Any idea how much larger this makes the library? It may be better to do as a separate driver.

**Comment by WoefulDerelict** [comment]
@tannewt I can definitely include some examples on how the extended API is used. Do you have any usage cases you'd specifically like documented? I can include a very basic example in the readme and more explorative items as things the users can execute in the examples folder.

I have not explored how much larger this makes the library. It not only adds support for periodic data acquisition mode it also includes support for configuring the sensor's behaviour in single shot data acquisition mode as a side effect. 

This represents a half step in my own efforts to generate a CircuitPython compatible driver for the SHT3X-D family of sensors that is capable of manipulating the alert limits and allowing users to explore these features of the platform including the alert pin on the breakout board. The alert limits only trip when the sensor is in periodic data acquisition mode so properly exposing that feature set required building out this one. As this does represent a rather massive divergence from the original driver which only supported high repeatability single shot data acquisition mode without clock stretching specifically on the SHT31-D breakout there may be an argument for considering it a separate driver.

File: adafruit_sht31d.py [hunk 1]
```diff
@@ -57,18 +58,67 @@
...
+_SHT31_CLEARSTATUS = const(0x3041)
+_SHT31_HEATER_ENABLE = const(0x306D)
+_SHT31_HEATER_DISABLE = const(0x3066)
+_SHT31_SOFTRESET = const(0x30A2)
+_SHT31_NOSLEEP = const(0x303E)
+_SHT31_PDA_FETCH = const(0xE000)
+_SHT31_PDA_BREAK = const(0x3093)
+
+MODE_SSDA = 'Single'
+MODE_PDA = 'Periodic'
```
Comment (tannewt) [inline adafruit_sht31d.py]: Please rename these `MODE_SINGLE` and `MODE_PERIODIC` and then use them in the examples. Using strings prevents you from changing the value without breaking the API.

**Review by tannewt — CHANGES_REQUESTED** [review]
The example looks good. Thanks! One more comment on using the string values for mode in the examples. Should be good after that!

**Review by tannewt — APPROVED** [review]
Looks good! Thank you!

File: adafruit_sht31d.py (hunk 1 above)
Comment (WoefulDerelict) [reply adafruit_sht31d.py]: I've renamed the variables and updated the example as requested. I've also included an example that exercises more of the single shot data acquisition mode features.

**Comment by tannewt** [comment]
Thank you for the second example!

**Pull request merged**